Firestore/Example/Tests/Local/FSTQueryCacheTests.mm

```diff
@@ -56,242 +56,267 @@ - (BOOL)isTestBaseClass {
 - (void)testReadQueryNotInCache {
   if ([self isTestBaseClass]) return;
 
+  FSTWriteGroup *group = [self.persistence startGroupWithAction:@"ReadQueryNotInCache"];
```

Comment:
Why not

```objc
[self.persistence runTransactionWithAction:@"ReadQueryNotInCache" block:^{
  // ...
}];
```

Here and throughout.

Reply:
As above. Without everything being transitioned at once, you are limited in where you can use transaction blocks, and I think it makes the most sense to do that switch in one pass at the end.